src/sql/base/browser/ui/table/plugins/headerFilter.plugin.ts

```diff
@@ -337,55 +337,71 @@ export class HeaderFilter<T extends Slick.SlickData> {
 	}
 
 	private getFilterValues(dataView: Slick.DataProvider<T>, column: Slick.Column<T>): Array<any> {
-		const seen: Array<string> = [];
-		for (let i = 0; i < dataView.getLength(); i++) {
-			const value = dataView.getItem(i)[column.field!];
-
-			if (!seen.some(x => x === value)) {
-				seen.push(value);
+		const seen: Set<string> = new Set();
+		dataView.getItems().forEach(items => {
+			const value = items[column.field!];
+			if (value instanceof Array) {
+				value.forEach(v => seen.add(v));
+			} else {
+				seen.add(value);
 			}
-		}
-		return seen;
+		});
+
+		return Array.from(seen);
 	}
 
 	private getFilterValuesByInput($input: JQuery<HTMLElement>): Array<string> {
 		const column = $input.data('column'),
 			filter = $input.val() as string,
 			dataView = this.grid.getData() as Slick.DataProvider<T>,
-			seen: Array<any> = [];
+			seen: Set<any> = new Set();
+
+		dataView.getItems().forEach(item => {
+			const value = item[column.field];
+			if (value instanceof Array) {
+				if (filter.length > 0) {
```

Comment:
Instead of duplicating the logic here I think it'd be better to just have the logic always take in an array and then just do something like 

`const valueArr = value instanceof Array ? value : [value];`;